### PR TITLE
Add licenseIdMigrationMessages. Resolves #5109.

### LINKED ISSUE #5109: license: BSD3 not recognized
Needs to be rewritten to `BSD-3-Clause`.

That can't be right.

Error is:
```
unexpected Unknown SPDX license identifier: BSD3
```

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: boot/SPDX.LicenseId.template.hs [hunk 1]
```diff
@@ -33,14 +35,53 @@ instance Binary LicenseId
...
+-- >>> traverse eitherParsec ["GPL-2.0", "GPL-3.0", "LGPL-2.1", "MIT", "ISC", "MPL-2.0", "Apache-2.0"] :: Either String [LicenseId]
+-- Right [GPL_2_0,GPL_3_0,LGPL_2_1,MIT,ISC,MPL_2_0,Apache_2_0]
+--
+licenseIdMigrationMessage :: String -> String
+licenseIdMigrationMessage = go where
+    go "BSD3"              = "Do you mean BSD-3-Clause?"
+    go "BSD2"              = "Do you mean BSD-2-Clause?"
+    go "AllRightsReserved" = "You can use NONE as a value of license field."
+    go "OtherLicense"      = "SPDX license list contains plenty of licenses. See https://spdx.org/licenses/. Also they can be combined into complex expressions with AND and OR."
+    go "PublicDomain"      = "Public Domain is a complex matter. See https://wiki.spdx.org/view/Legal_Team/Decisions/Dealing_with_Public_Domain_within_SPDX_Files. Consider using proper license."
```
Comment (gbaz) [inline boot/SPDX.LicenseId.template.hs]: Add the indefinite article: "Consider using a proper license." :-)

Comment (hvr) [reply boot/SPDX.LicenseId.template.hs]: That's good idea!

**Comment by phadej** [comment]
For the record, `PublicDomain` is hard, but my IANAL-understanding:

```
copyright: Public Domain
license: CC0
```

is what you'd write. (In my jurisdiction even I write so, I'll still retain copyright!)

https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses#Approvals
- CC0 isn't approved or rejected by OSI, Recommended by FSF (*If you want to release your work to the public domain, we recommend you use CC0.*: https://www.gnu.org/licenses/license-list.en.html#CC0)
- Unlicense has partial approval by OSI: while considered free not recommended
- Both are FSF approved and compatible with GPL-3.0

**Comment by phadej** [comment]
I'm not sure what happens on travis
```
/home/travis/build/haskell/cabal/dist-newstyle/src/pretty-show-1.6.16/dist: renamePath:rename: unsatisfied constraints (Directory not empty)Configuring library for pretty-show-1.6.16.
```

**Comment by angerman** [comment]
@phadej I keep seeing that as well. Initially I thought I'd blame the basedir patch (#4874), but as that hasn't been merged yet. I'm also a bit confused.

What apparently happens is that there `Pretty.Text.Parser` module (generated by alex/happy I believe) ends up in `pretty-show/dist/build`, and for some reason, cabal tries to rename the `dist` part (it's in `cabal-install`s `Install.hs` I believe). Not sure why we try to rename it. But that somehow trips up in System.Directory and ends up throwing a Directory not Empty exception.

However, while I occationally can reproduce this locally, when I try to run it verbosely (`-v`), it doesn't happen.

**Comment by angerman** [comment]
PS: restarting travis jobs might help (and will eventually turn green...)

**Pull request merged**